blob/gcsblob/example_test.go

```diff
@@ -64,3 +65,27 @@ func Example_open() {
 	_ = b
 	_ = err
 }
+
+func Example_As() {
```

Comment:
So what's the overall strategy you're proposing here? So far, we've been only including examples for how to open a bucket along with each driver, and putting examples of how to use the concrete type functions in the concrete type package. `As` is a weird one because it's kind of a mix. Are you suggesting we should add a similar example to every driver? Just for top-level `As`, or for each subtype? (both of those feel like overkill to me, I'd rather have a good example in the concrete type).

Reply:
I'm not proposing a strategy, just adding an example where none existed before. I agree repeating this many times is an overkill - but having it in one place gives us something working to point to. So far the example in `blob` only shows something where `As` returns `false`; I figured that at least one place where it returns `true` is useful.

Reply:
Why not add this in the concrete type then?

Reply:
I don't mind strongly either way. Wanted to keep provider-specific stuff from the generic `blob` package doc, but if it's not a concern I can move it there

Reply:
I suggest replacing the existing `As` example in the concrete type (which, as you point out, it poor because it is expected to fail) with this one, which is a better example. It might also be good to extend it a bit to include `As` for write, read, error, and maybe even list.

Be sure to point out that the specific types used for each are documented in the driver-specific package doc (e.g., link to https://godoc.org/gocloud.dev/blob/gcsblob#hdr-As as an example).

Reply:
PTAL. Moved it.
I actually think a single example is sufficient since they're all going to be pretty much the same.
The only one different is using `ErrorAs`, but this will likely require to use the s3 blob - I don't think gcsblob actually returns the gcs-specific error type anywhere